Source/WebCore/style/CustomPropertyRegistry.cpp

```diff
@@ -194,5 +188,50 @@ void CustomPropertyRegistry::notifyAnimationsOfCustomPropertyRegistration(const
     }
 }
 
+auto CustomPropertyRegistry::parseInitialValue(const Document& document, const AtomString& propertyName, const CSSCustomPropertySyntax& syntax, CSSParserTokenRange tokenRange) -> Expected<std::pair<RefPtr<CSSCustomPropertyValue>, ViewportUnitDependency>, ParseInitialValueError>
+{
+    // FIXME: This parses twice.
+    auto dependencies = CSSPropertyParser::collectParsedCustomPropertyValueDependencies(syntax, tokenRange, strictCSSParserContext());
+    if (!dependencies.isComputationallyIndependent())
+        return makeUnexpected(ParseInitialValueError::NotComputationallyIndependent);
+
+    // We don't need to provide a real context style since only computationally independent values are allowed (no 'em' etc).
+    auto placeholderStyle = RenderStyle::create();
+    Style::Builder builder { placeholderStyle, { document, RenderStyle::defaultStyle() }, { }, { } };
```

Comment:
It seems we need a new static constructor `Style::Builder::makeWithoutStyleContext()` ?

Reply:
The constructor takes a reference to a `RenderStyle`, which is not a refcounted object. So we'd have to have a `RenderStyle` embedded in / owned by the `Builder` itself for cases like this. I'm going to leave this as is.